### ISSUE TITLE
Feature Request: File deleting

### ISSUE THREAD

**Issue by kalamazandy** [gopro/labs#291]
I've got the GoPro in a box recording time-lapse footage, using a CamDo UpBlink (they had be believe their system could delete files but it can't, probably because of this)

For safety concerns, we could only allow this if a certain file exists on the SD card, like create a "QR_del_OK" extensionless file on the root of the card to allow the feature? 

Now I can send delete commands to the GoPro. Deleting all images, previews, videos, etc at once would be one option. Wouldn't be able to send a format command if the file to allow the option exists, but if there's a way to handle that, then that would work as well. 
Options to delete with instructions would be nice as well. Maybe delete all files older than "XX" days would work well.
I'm sure some people might appreciate delete by type, like delete all preview files, or all images. I'd like to be able to take Only raw images, so if that's not an option (I haven't searched out if that's possible through labs yet), so delete jpeg images but leave the raw. Also could be good if I accidentally set it up to save both, but didn't want the jpegs or raw images. 

Basically, it would be cool to be able to clear off media from the SD card, setting various options to do so. That would be very helpful.

**Comment by dnewman-gpsw** [comment]
There have been some steps towards this, because applications like dash cam need to make sure there's room on the SD card.  In the latest HERO11 Mini build is a command that allows you to delete up to half the SD card, always deleting the oldest media first. This is still security minded, attempting to limit damage from malicious QR Codes.  Unlike !FORMAT, this does not require a user approval prompt, so it can be used in a script.  This will be in HERO11 later in the year, and likely also MAX and HERO10, there are currently Labs scheduled updates for other cameras. 

From the release notes:
  - Added !Cx where x is GBs to make sure is clear. If more than x is available, no files are removed. x is limited to half SD size. Deletes the oldest media first. Ideal for dashcam applications.

**Comment by dnewman-gpsw** [comment]
P.S. Your QR_Del_OK, is a great idea.  That could be a way to bypass user prompting if used in a script.
